### ISSUE TITLE
Type on a specific window?

### ISSUE THREAD

**Issue by GGutriedGG** [boppreh/keyboard#360]
Can I make my script type on a specific window, even when its minimized? So I don't have to have the window in front of me the whole time and I can just do something else.

**Comment by Mr-oy** [comment]
type your script in a  '.py' file , after testing your script ,you can just rename it to '.pyw' file.
the python will then try to show you the gui windows , however your script doesn't contain any gui window . That means you won't see anything , and the script just working background.

**Comment by gavinkrezel** [comment]
I don't think that quite answered the question, @Mr-oy , but it is impossible to what you are asking @GGutriedGG because it is actually 'typing'. What I did though was I set up a virtual machine that would run a typing bot in the background while I did whatever else I wanted with my computer.

**Comment by Mr-oy** [comment]
Sry, I'm new here.

I thought that @GGutriedGG had some problem with running the script in background progress , which may not so easy in windows.

Actually I know nearly nothing about the virtual machine. Do you mind if I ask you @gavinkrezel  how to get the keyboard event in the real machine while the typing bot ,or rather, the script is actually running in the virtual one.   Thank you very much.

**Comment by gavinkrezel** [comment]
Well, you can create a virtual machine by using Oracle VirtualBox. I created a Lubuntu VM (Lightweight Ubuntu), installed python3 on it and downloaded the keyboard package onto it. I moved those to the proper folder and bam, it works. Now I just have a script that makes my Discord account chat in a server every once in a while for some stupid game that my friends and I have, which runs in the background of my actual OS, allowing me to do whatever I want without worrying about turning the bot off.

**Comment by Mr-oy** [comment]
Thank you ,i'll try the Oracle VirtualBox. By the way , Is it a kind of Docker?

**Comment by GGutriedGG** [comment]
> type your script in a '.py' file , after testing your script ,you can just rename it to '.pyw' file.
> the python will then try to show you the gui windows , however your script doesn't contain any gui window . That means you won't see anything , and the script just working background.

I mean i have a python script that opens a selenium browser and i want to be able to minimize the browser while the script is still typing. There is no python gui, I run the script via c# code.

**Comment by gavinkrezel** [comment]
@GGutriedGG yeah, I wish it would work like that, however it is actually making the computer think that your actual peripherals are sending the input, meaning it is like your own keyboard is sending it. That is why I just used a virtual machine in the background of my computer. 

I guess a better way to say this is to think about using your own keyboard. You can't send keyboard input to a game while simultaneously typing in a text document. It is the same general concept here.

**Comment by gavinkrezel** [comment]
also @Mr-oy I am not entirely sure. You could probably look it up and find out pretty quickly though :)
